File: 02_config/app/main.nim
import asynchttpserver, asyncdispatch
from strutils import parseInt
from os import getEnv

var server = newAsyncHttpServer()

proc callback(req: Request) {.async.} =
  let data = readFile("/etc/app/config.json")
  await req.respond(Http200, "Hello Nim\n" & data)

let port = getEnv("SERVER_PORT").parseInt
waitFor server.serve(Port(port), callback)

File: 02_config/app/main.go
package main

import (
  "os"
  "io/ioutil"
  "strconv"
  "fmt"
  "net/http"
)

func handler(w http.ResponseWriter, r *http.Request) {
  fmt.Fprintf(w, "Hello, World")

  b, err := ioutil.ReadFile("/app/data.json")
  if err != nil {
    panic(err)
  }
  fmt.Fprintf(w, string(b))
}

func main() {
  port := os.Getenv("SERVER_PORT")

  // validation
  _, err := strconv.Atoi(port)
  if err != nil {
    panic(err)
  }

  http.HandleFunc("/", handler)
  http.ListenAndServe(":" + port, nil)
}